src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -31,39 +51,43 @@ public String todayDate(DateTimePart datePart) {
      *                   - 3-й элемент массива - день (число);
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
+        } catch (Exception e) {
+            return Optional.empty();
```

Comment:
Good, but let's move it after catch block

Reply:
we should not use try catch as if else (build our logic on it)

Reply:
If i wont use "try-catch" as "if-else" here i definitely should implement new method "isDateValid()" and use "try-catch" as "if-else" in it to check validity of a date.

Reply:
you can just put return after catch block. And let's specify catching to these exceptions DateTimeException | ArrayIndexOutOfBoundsException e